spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -224,4 +224,130 @@
       }.to raise_error(WhitehallImporter::AbortImportError)
     end
   end
+
+  context "when the document is scheduled" do
+    let(:future_date) { Time.zone.now.tomorrow.rfc3339 }
+    let(:document) { create(:document, imported_from: "whitehall", locale: "en") }
+    let(:whitehall_edition) do
+      build(
+        :whitehall_export_edition,
+        state: "scheduled",
+        revision_history: [
+          build(:revision_history_event),
+          build(:revision_history_event, event: "update", state: "submitted"),
+          build(:revision_history_event, event: "update", state: "scheduled"),
+        ],
+        scheduled_publication: future_date,
+      )
+    end
+
+    let(:edition) do
+      described_class.call(document: document, whitehall_edition: whitehall_edition)
+    end
```

Comment:
We should try avoid the practice we've used here where the code that is being under test is called outside of the test. As the effect of it is that when you're reading the tests you're having to go off to other places to work out where the code that is being tested is called. 

It looks like you followed precedent from withdraw here which unfortunately already had this pattern whereas Pete didn't use that for unpublishing.